### PR TITLE
Adding AzureRM lib and scale-set support

### PR DESCRIPTION
**Pull request by lshahar**

### CODE REVIEW

**Comment by LindsayHill** [comment]
Thanks for the PR!

I've just made a separate change to implement `config.schema.yaml` for this pack. Could you rebase your PR from master, update your changes to incorporate `config.schema.yaml` and `azure.yaml.example`, and increment the pack version?

File: README.md [hunk 1]
```diff
@@ -49,6 +49,8 @@ You can also use dynamic values from the datastore. See the
 
 * ``create_resource`` - Create a generic ARM resource
 * ``create_linked_resource_url`` - Create a linked (template and parameter) resource from a URI
+* ``list_resource_groups`` - List the names of the resource groups
+* ``default_resource_group`` - Default resource group
```
Comment (LindsayHill) [inline README.md]: That section is a list of actions - I'm not sure that `default_resource_group` belongs there?

**Comment by LindsayHill** [comment]
Looks like it still needs an update to `config.schema.yaml` - probably a section at the end that looks like this:

```yaml
      default_resource_group:
        description: "Default Resource Group"
        type: "string"
        required: false
```

Should go at the end, with same indentation as item above.

**Comment by LindsayHill** [comment]
Is this now ready for merge? I think the only thing outstanding is to increment `pack.yaml` and CHANGES.md

**Comment by lshahar** [comment]
Done, can you approve it ?

**Comment by LindsayHill** [comment]
Just needs .idea/stackstorm-azure.iml removed - can you remove that from the PR?

**Comment by lshahar** [comment]
Done

**Pull request merged**